### PR TITLE
Levelview observer pattern

### PR DESCRIPTION
**Pull request by hlycklama**
Major refactoring of Level

### CODE REVIEW

**Comment by jorisoudejans** [comment]
Maybe in the future we should make an abstract Level, but I can see the advantages of just extending the normal Level for SurvivalLevel. @cbilstra @JaspervanTilburg @skoppers What do you think?

**Comment by JaspervanTilburg** [comment]
I agree on that. Good mvc building!

File: src/main/java/doob/model/SurvivalLevel.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            spawnBalls(System.currentTimeMillis());
+        }
+    }
+
+    /**
+     * Adds one to time.
+     * @return one
+     */
+    @Override
+    protected int getTimeMutation() {
```
Comment (skoppers) [inline src/main/java/doob/model/SurvivalLevel.java]: What does this do, exactly?

Comment (jorisoudejans) [inline src/main/java/doob/model/SurvivalLevel.java]: Haha you did that, right? Can you explain? Has something to do with survival I think.

Comment (skoppers) [inline src/main/java/doob/model/SurvivalLevel.java]: I meant the getTimeMutation() function

Comment (jorisoudejans) [inline src/main/java/doob/model/SurvivalLevel.java]: Ah, that is the function that gets 1 or -1, depending on whether it is called from Level or SurvivalLevel. This way, we don't have to implement update() twice, just override the getTimeMutation method.

Comment (skoppers) [inline src/main/java/doob/model/SurvivalLevel.java]: Ah, that makes sense!

**Comment by skoppers** [comment]
I do think it would be better to make an abstract Level class, although this now works, you shouldn't construct a hierarchy for the sole reason of code re-use.
Doesn't this also break the Liskov substitution principle?

**Comment by jorisoudejans** [comment]
You were absolutely right. Changed it.

**Pull request merged**